gtk/src/toga_gtk/widgets/canvas.py

```diff
@@ -47,55 +87,86 @@ def gtk_draw_callback(self, widget, cairo_context):
         """
 
         # Explicitly render the background
-        sc = self.native.get_style_context()
-        bg = sc.get_property("background-color", sc.get_state())
-        cairo_context.set_source_rgba(
-            255 * bg.red,
-            255 * bg.green,
-            255 * bg.blue,
-            bg.alpha,
-        )
-        width = self.native.get_allocation().width
-        height = self.native.get_allocation().height
-        cairo_context.rectangle(0, 0, width, height)
-        cairo_context.fill()
+        if self._background_color and not self._background_color == TRANSPARENT:
```

Comment:
Is this a required part of the implementation, or an optimisation you've found along the way? It strikes me as odd that drawing a transparent background would be different to *not* drawing a background.

Reply:
Ok - but the "optimization" requires adding a shadow representation of background color - which means there's a variable that *could* get out of sync. 

Handling the "special case" of transparency isn't that onerous - `TRANSPARENT` *does* have a clear RGBA representation. If avoiding the optimisation means we no longer need that shadow variable, I'd prefer to see the "non-optimised" version.